full-stack-ruby-on-rails/project-binary-search-trees/lib/tree.rb

```diff
@@ -0,0 +1,195 @@
+# frozen_string_literal: true
+
+require_relative './node'
+
+# A class which constructs a binary search tree from an array
+class Tree
+  attr_reader :root
+
+  def initialize(array = [])
+    @root = build_tree(array)
+  end
+
+  def build_tree(array)
+    array = sanitize_array(array)
+    build_branches(array, start_index: 0, end_index: array.size-1)
+  end
+
+  def build_branches(array, start_index:, end_index:)
+    return nil if start_index > end_index
```

Comment:
This is typically what I see (no explicit `nil`):

```suggestion
    return if start_index > end_index
```

Reply:
NBD